packages/php-wasm/compile/php/phpwasm-emscripten-library.js

```diff
@@ -313,10 +311,51 @@ const LibraryExample = {
 			return 0;
 		}
 
+		let argsArray = [];
+
+		if (argsLength) {
+			var ptr = args;
+			// Extracts an array of CLI arguments that should be passed to `command`.
+			// On the C side, the args are expressed as `**char` so we must go read
+			// each of the `argsLength` `*char` pointers and convert the associated data into
+			// a JavaScript string.
+			for (var i = 0; i < argsLength; i++) {
+				var str = UTF8ToString(ptr);
+				// The exact offset between two arguments, depends on the length of the actual argument.
+				// The minimum offset is 16. If the string length is above 16, the correct offset has to be a multiple of 8.
```

Comment:
Thank you for explaining! What is the source of that heuristics? Is there any chance it would break with different compilation flags? Or is it stable and just how the memory layout works in C?